### DISCUSSION TITLE
How to use laserscan tool

### DISCUSSION THREAD

**Discussion by orient755** [OpenOrienteering/mapper#1762] [Questions & Answers] [answered]
Pls, tell me, how to use laserscan tool? The programme return ERROR.
![laserscan error](https://user-images.githubusercontent.com/73453265/97128412-9f8c7580-177f-11eb-9869-451e0fa91603.jpg)

**Comment by puzzlepaint** [comment, marked as answer]
The likely reason for this error is that the program assumes a different class number for ground points than what is actually used in the given input file, such that it does not find any and passes zero vertices to the triangulation mentioned above the error message. Unfortunately, in the released version from 2012, the tool's option to change the ground class (`-las-ground <x>`) has no effect due to a bug. Given that the lasertool program is quite old and has not been well tested, I would thus recommend looking into other tools to process the data, unless you would be willing to change the tool's source code.

**Comment by dg0yt** [comment]
Alternative free laser data tools:
- http://www.routegadget.net/karttapullautin/
- http://oapp.se/Applikationer/OL_Laser.html
- http://tmsw.no/mapping/basemap_generation.html
